src/ParametersHandler/tests/ParametersHandlerTest.cpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file ParametersHandlerTest.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+// std
+#include <any>
+#include <iostream>
+#include <memory>
+#include <unordered_map>
+
+// Catch2
+#include <catch2/catch.hpp>
+
+#include <BipedalLocomotionControllers/ParametersHandler/IParametersHandler.h>
+
+using namespace BipedalLocomotionControllers::ParametersHandler;
+
+// example of parameters handler
+class BasicImplementation : public IParametersHandler<BasicImplementation>
+{
+    std::unordered_map<std::string, std::any> m_map;
+
+public:
+    BasicImplementation(const std::unordered_map<std::string, std::any>& map)
+        : m_map{map}
+    {}
+
+    BasicImplementation() = default;
+
+    template <typename T> void setParameter(const std::string& parameterName, const T& parameter)
+    {
+        m_map.insert({parameterName, std::make_any<T>(parameter)});
+    }
+
+    template <typename T> bool getParameter(const std::string& parameterName, T& parameter) const
+    {
+        auto parameterAny = m_map.find(parameterName);
+        if (parameterAny == m_map.end())
+        {
+            std::cerr << "[BasicImplementation::getParameter] Parameter named " << parameterName
+                      << " not found." << std::endl;
+            return false;
+        }
+
+        try
+        {
+            parameter = std::any_cast<T>(parameterAny->second);
+        } catch (const std::bad_any_cast& exception)
+        {
+            std::cerr << "[BasicImplementation::getParameter] The type of the parameter named "
+                      << parameterName << " is different from the one expected" << std::endl;
+            return false;
+        }
+
+        return true;
+    }
+
+    std::unique_ptr<IParametersHandler<BasicImplementation>> getGroup(const std::string& name) const
+    {
+        auto group = m_map.find(name);
+        if (group == m_map.end())
+            return std::make_unique<BasicImplementation>();
+
+        std::unordered_map<std::string, std::any> map;
+        try
+        {
+            map = std::any_cast<decltype(map)>(group->second);
+        } catch (const std::bad_any_cast& exception)
+        {
+            std::cerr << "[BasicImplementation::getGroup] The element named " << name
+                      << " is not a 'std::unordered_map<std::string, std::any>'" << std::endl;
+            return std::make_unique<BasicImplementation>();
+        }
+
+        return std::make_unique<BasicImplementation>(map);
+    }
+
+    friend std::ostream& operator<<(std::ostream& os, const BasicImplementation& handler)
```

Comment:
Why is the ``friend`` required? Wouldn't it be easier with a ``toString()`` method?

Reply:
As discussed F2F the `toString()` function has been implemented in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/13